app/main.py

```diff
@@ -1 +1,35 @@
-# write your code here
+class Animal:
+
+    alive = []
+
+    def __init__(
+            self,
+            name: str,
+            health: int = 100,
+            hidden: bool = False
+    ) -> None:
+        self.name = name
+        self.health = health
+        self.hidden = hidden
+        Animal.alive.append(self)
+
+    def __repr__(self) -> str:
+        return f"{{Name: {self.name}, " \
+               f"Health: {self.health}, " \
+               f"Hidden: {self.hidden}}}"
+
+
+class Herbivore(Animal):
+    def hide(self) -> None:
+        self.hidden = not self.hidden
+
+
+class Carnivore(Animal):
+    @staticmethod
+    def bite(target: Herbivore) -> None:
+        if not isinstance(target, Carnivore) and not target.hidden:
+            if target.health > 50:
+                target.health -= 50
+            elif target.health <= 50:
+                target.health = 0
+                Animal.alive.pop(Animal.alive.index(target))
```

Comment:
You can use just one "remove" method, that allows you to delete object from the list by itself, instead of using pop and index methods.